Lib/test/test_pickletools.py

```diff
@@ -62,6 +63,315 @@ def test_optimize_binput_and_memoize(self):
         self.assertNotIn(pickle.BINPUT, pickled2)
 
 
+class SimpleReader:
+    def __init__(self, data):
+        self.data = data
+        self.pos = 0
+
+    def read(self, n):
+        data = self.data[self.pos: self.pos + n]
+        self.pos += n
```

Comment:
Probably not applicable, but what if `self.pos+n < len(self.data)`?

Reply:
You perhaps mean `self.pos+n > len(self.data)`? We will get truncated data, and the following reads will return an empty bytes object. Like in real files or BytesIO.

SimpleReader is not currently used in such tests, because it does not make difference, but I am sure that tests for truncated data would pass with it.

Reply:
Perfect then.